operator/internal/common/extensions/validate_deployment_defaults_test.go

```diff
@@ -0,0 +1,112 @@
+package extensions
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	platform "github.com/stackrox/rox/operator/api/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+)
+
+func TestValidateDeploymentDefaultsExtension(t *testing.T) {
+	tests := map[string]struct {
+		obj         *unstructured.Unstructured
+		expectError bool
+	}{
+		"valid configuration": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: false,
+		},
+		"invalid: pinToNodes with nodeSelector": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+			})),
+			expectError: true,
+		},
+		"invalid: pinToNodes with tolerations": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: true,
+		},
+		"invalid: pinToNodes with both nodeSelector and tolerations": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: true,
+		},
+		"skips validation for deleted objects": {
+			obj: func() *unstructured.Unstructured {
+				central := centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+					PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+					NodeSelector: map[string]string{
+						"global-node-selector-label": "global-node-selector-value",
+					},
+				})
+				now := metav1.NewTime(time.Now())
+				central.SetDeletionTimestamp(&now)
+				return toUnstructured(t, central)
+			}(),
+			expectError: false,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ext := ValidateDeploymentDefaultsExtension()
+			err := ext(context.Background(), tt.obj, nil, logr.Discard())
```

Comment:
Perhaps we can start using:
```suggestion
			err := ext(t.Context(), tt.obj, nil, logr.Discard())
```